Makefile

```diff
@@ -21,72 +21,69 @@ RUST_ACTION ?= run -p sigstore-protobuf-specs-codegen
+	${DOCKER_RUN} ${PROTOC_IMAGE} \
+		-d /defs/protos \
+		-o /defs/gen/pb-go \
+		-l go --go-module-prefix github.com/sigstore/protobuf-specs/gen/pb-go
 
 python: docker-image
 	@echo "Generating python protobuf files"
 # we need to manually fix the PYTHONPATH due to: https://github.com/namely/docker-protoc/pull/356
-	docker run \
-		--platform linux/amd64 \
-		-v ${PWD}:/defs \
-		-e PYTHONPATH="/opt/mypy-protobuf/" \
+	${DOCKER_RUN} -e PYTHONPATH="/opt/mypy-protobuf/" \
 		--entrypoint bash ${PROTOC_IMAGE} \
-		-c "cd ./gen/pb-python/sigstore_protobuf_specs && protoc -I/opt/include -I../../../protos/ --python_betterproto_out=. ../../../protos/*.proto"
+		-c "cd /defs/gen/pb-python/sigstore_protobuf_specs && protoc -I/opt/include -I../../../protos/ --python_betterproto_out=. ../../../protos/*.proto"
 
 typescript: docker-image
 	@echo "Generating javascript protobuf files"
-	docker run \
-		--platform linux/amd64 \
-		-v ${PWD}:/defs \
-		${PROTOC_IMAGE} \
-		-d protos -l typescript -o ./gen/pb-typescript/src/__generated__ --ts_opt oneof=unions,forceLong=string,env=node,exportCommonSymbols=false,outputPartialMethods=false,outputEncodeMethods=false,unrecognizedEnum=false
+	${DOCKER_RUN} ${PROTOC_IMAGE} \
+		-d /defs/protos -l typescript \
+		-o /defs/gen/pb-typescript/src/__generated__ \
+		--ts_opt oneof=unions,forceLong=string,env=node,exportCommonSymbols=false,outputPartialMethods=false,outputEncodeMethods=false,unrecognizedEnum=false
 
 ruby: docker-image
 	@echo "Generating ruby protobuf files"
-	docker run \
-		--platform linux/amd64 \
-		-v ${PWD}:/defs \
-		--entrypoint bash ${PROTOC_IMAGE} \
-		-c "cd ./gen/pb-ruby && protoc -I/opt/include -I../../protos/ --ruby_out=lib ../../protos/*.proto"
+	${DOCKER_RUN} --entrypoint bash ${PROTOC_IMAGE} \
+		-c "cd /defs/gen/pb-ruby && protoc -I/opt/include -I../../protos/ --ruby_out=lib ../../protos/*.proto"
 
 jsonschema: docker-image-jsonschema
 	@echo "Generating JSON schema files"
-	docker run \
-	       -v ${PWD}:/defs \
-	       --entrypoint sh \
-	       ${JSONSCHEMA_IMAGE} \
-	       -c "cd defs/gen/jsonschema && ./jsonschema.sh -I ../../protos -I /googleapis/ --jsonschema_out=schemas ../../protos/*.proto"
+	${DOCKER_RUN} --entrypoint sh \
+		${JSONSCHEMA_IMAGE} \
+		-c "cd /defs/gen/jsonschema && ./jsonschema.sh -I ../../protos -I /home/builder/googleapis/ --jsonschema_out=schemas ../../protos/*.proto"
 
 rust: docker-image
-	docker run \
-		--platform linux/amd64 \
-		-v ${PWD}:/defs \
-		-e "RUST_BACKTRACE=1" \
+	@echo "Generating rust protobuf files"
+	${DOCKER_RUN} -e "RUST_BACKTRACE=1" \
 		-e "CARGO_REGISTRY_TOKEN" \
 		--entrypoint bash ${PROTOC_IMAGE} \
-		-c "cd gen/pb-rust && cargo ${RUST_ACTION}"
+		-c "cd /defs/gen/pb-rust && cargo ${RUST_ACTION}"
 
 # docker already does its own caching so we can attempt a build every time
 .PHONY: docker-image
 docker-image:
 	@echo "Building development docker image"
-	docker build -t ${PROTOC_IMAGE} .
+	${DOCKER_BUILD} -t ${PROTOC_IMAGE} .
```

Comment:
If you pass in the current host system user and group as args here you can use those IDs when creating the user and group in the docker file instead of 1000

Reply:
That's actually what I'm doing. They are passed as build-args, and if provided they are used instead of the default value of 1000.

Reply:
Or only the user id to be exact. But it should be enough.